README.md

```diff
@@ -22,82 +22,201 @@ yarn add -S react-plaid-link
 Please refer to the [official Plaid Link docs](https://plaid.com/docs/#creating-items-with-plaid-link) for
 a more holistic understanding of the various Link options.
 
-### Examples
+## Examples
 
-Head to the `react-plaid-link` [storybook](https://plaid.github.io/react-plaid-link) to try it out for yourself, or
-checkout:
-
-- [Storybook](https://plaid.github.io/react-plaid-link)
-- [`/examples/hooks.js`](./examples/hooks.js)
-- [`/examples/hoc.js`](./examples/hoc.js)
+Head to the `react-plaid-link` [storybook](https://plaid.github.io/react-plaid-link) to try out a live demo.
 
 ### Using React hooks
 
-This is the new and preferred approach for integrating with Plaid Link in React.
+This is the preferred approach for integrating with Plaid Link in React.
 
-```jsx
-import React, { useCallback } from 'react';
-import { usePlaidLink } from 'react-plaid-link';
+ℹ️ Note: `link_token` cannot be null when passed to `usePlaidLink`. Generate your link_token
+outside of the component where the hook is initialized.
 
-const App = () => {
-  const onSuccess = useCallback((token, metadata) => {
-    // send token to server
-  }, []);
+```tsx
+import React, { useCallback, useState, FunctionComponent } from "react";
+import {
+  usePlaidLink,
+  PlaidLinkOptions,
+  PlaidLinkOnSuccess,
+} from "react-plaid-link";
+
+interface Props {
+  token: string;
+}
 
-  const config = {
-    token: '<GENERATED_LINK_TOKEN>',
+const PlaidLink: FunctionComponent<Props> = ({ token }) => {
+  const onSuccess = useCallback<PlaidLinkOnSuccess>(
+    (public_token, metadata) => {
+      // send public_token to server
+    },
+    []
+  );
+
+  const config: PlaidLinkOptions = {
+    token,
     onSuccess,
-    // ...
+    // onExit
+    // onEvent
   };
 
   const { open, ready, error } = usePlaidLink(config);
 
   return (
-    <MyButton onClick={() => open()} disabled={!ready}>
+    <button onClick={() => open()} disabled={!ready}>
       Connect a bank account
-    </MyButton>
+    </button>
+  );
+};
+
+const App: FunctionComponent = () => {
+  const [token, setToken] = useState<string | null>(null);
+
+  // generate a link_token
+  React.useEffect(() => {
+    async function createLinkToken() {
+      let response = await fetch("/api/create_link_token");
+      const { link_token } = await response.json();
+      setToken(link_token);
+    }
+    createLinkToken();
+  }, []);
+
+  // only initialize Link once our token exists
+  return token === null ? (
+    <div className="loader"></div>
```

Comment:
is it worth leaving a comment here of "insert your loading animation here" OR plugging our loader from `react-threads` the `loadingspinner`?